srcs/requirements/frontend/src/events.ts

```
@@ -0,0 +1,29 @@
+ * @brief Sets up a listener for the popstate event.
+ * 
+ * This function adds an event listener to the window object to handle the popstate event.
+ * When the user navigates through the browser history, this listener ensures that the
+ * application navigates to the correct view stored in the history state.
+ */
+
+export function setupHistoryListener(): void {
+    window.addEventListener("popstate", (event) => {
+        const state = event.state;
+
+        if (state && state.view) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)